### PR TITLE
feat: auto merge (#33)

### PR DESCRIPTION
**Pull request by Reimei1213**
- add auto merge
- #33
- ref: https://docs.github.com/ja/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ github-actions-merger is github actions that merges pull request with commit mes
       "pr_number": ${{ github.event.issue.number }}
       "comment": ${{ github.event.comment.body }}
       "mergers": 'na-ga,0daryo'
+      "is_auto_merge": false
```
Comment (peaceiris) [inline README.md:16]: The `enable_auto_merge` looks better.

File: README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ github-actions-merger is github actions that merges pull request with commit mes
       "pr_number": ${{ github.event.issue.number }}
       "comment": ${{ github.event.comment.body }}
       "mergers": 'na-ga,0daryo'
+      "enable_auto_merge": false
```
Comment (peaceiris) [inline README.md:16]: We need to add the enable_auto_merge to the inputs.

https://github.com/abema/github-actions-merger/blob/487288a05a33bdf87cadf3be2df074ea2a96e93f/action.yml#L10

File: README.md [hunk 3]
```diff
@@ -44,7 +44,16 @@ pr_number: ${{ github.event.issue.number }}
...
 mergers: 'comma separeted github usernames. every user is allowed if not specified'
+enable_auto_merge: true
 ```
 
+## Options
+### Enable Auto Merge
+- [About auto merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request#about-auto-merge)
+- You can use the auto merge when `enable_auto_merge` is true.
+- Default is `false`.
+- For more information about enabling auto merge to see the Note: [Enabling auto-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request#about-auto-merge).
```
[the remark this answers is not in the document]
Comment (peaceiris) [reply README.md:55]: Thanks!

File: main.go [hunk 4]
```diff
@@ -105,10 +108,16 @@ func (gh *ghClient) merge(ctx context.Context, owner, repo string, prNumber int,
...
 		return fmt.Errorf("failed to generate template: %w", err)
 	}
-	_, _, err = gh.client.PullRequests.Merge(ctx, owner, repo, prNumber, commitMsg, &github.PullRequestOptions{
-		CommitTitle: generateCommitSubject(pr),
-		MergeMethod: mergeMethod,
-	})
+
+	if enableAutoMerge {
+		// GitHub API docs: https://cli.github.com/manual/gh_pr_merge
+		err = exec.Command("gh", "pr", "merge", strconv.Itoa(prNumber), fmt.Sprintf("--%v", mergeMethod), "--auto", "--subject", generateCommitSubject(pr), "--body", commitMsg, "--repo", fmt.Sprintf("%v/%v", owner, repo)).Run()
```
Comment (0daryo) [inline main.go:114]: nits

```suggestion
		err = exec.Command("gh", "pr", "merge", strconv.Itoa(prNumber), fmt.Sprintf("--%v", mergeMethod), "--auto", "--subject", generateCommitSubject(pr), "--body", commitMsg, "--repo", fmt.Sprintf("%s/%s", owner, repo)).Run()
```

Comment (Reimei1213) [reply main.go]: https://github.com/abema/github-actions-merger/pull/36/commits/b31ffc1d4ee3955891003c8fe02fff79af54bae3#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R114

**Review by 0daryo — APPROVED** [review]

**Pull request merged**